### PR TITLE
Add logs according to Ori CR

### PR DESCRIPTION
**Pull request by yanaise**
Some changes to logs to help debugging of cache code
---
---
_Release Notes_: 
---
_User Notifications_:

### CODE REVIEW

**Comment by yanaise** [comment]
Screen Shot 2021-07-27 at 12 24 14 AM
Screen Shot 2021-07-27 at 12 27 36 AM

File: packages/workspace/src/workspace/nacl_files/elements_cache.ts [hunk 1]
```diff
@@ -325,17 +326,55 @@ export const createMergeManager = async (flushables: Flushable[],
...
+          const { elemID } = getChangeElement(change)
+          if (numDeleted < MAX_LOG_ON_MERGE) {
+            deletedIds.push(elemID.getFullName())
+          }
+          numDeleted += 1
+          return elemID
+        }))
+      if (numDeleted !== 0) {
+        log.debug(`Deleted ${numDeleted} elements. The first ${deletedIds.length} ids are: ${deletedIds.join(', ')}`)
+      }
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/elements_cache.ts:348]: why not functional? and why use `awu`?
I think this will be more readable as functional code
```typescript
const deletedIds = mergedChanges.changes.filter(isRemovalChange).map(change => getChangeElement(change).elemID)
await currentElements.deleteAll(deletedIds)
if (deletedIds.length > 0) {
  log....
}
```

File: packages/workspace/src/workspace/nacl_files/elements_cache.ts [hunk 2]
```diff
@@ -325,17 +326,55 @@ export const createMergeManager = async (flushables: Flushable[],
...
+          numModified += 1
+        }
+        return elem
+      }))
+    if (numModified !== 0) {
+      log.debug(`Modified ${numModified} elements. The first ${modifiedIds.length} ids are: ${modifiedIds.join(', ')}`)
+    }
+    if (numAdded !== 0) {
+      log.debug(`Added ${numAdded} elements. The first ${addedIds.length} ids are: ${addedIds.join(', ')}`)
+    }
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/elements_cache.ts:377]: Given that both add and modify get the same treatment in the code (they get `set`) I am not sure it is needed to separate them to different logs

Comment (yanaise) [reply packages/workspace/src/workspace/nacl_files/elements_cache.ts:377]: I thought it might have different implications in debug? Like if you're seeing adds when you expect modifications, it's possible that there's a problem in calculating changes?

Comment (ori-moisis) [reply packages/workspace/src/workspace/nacl_files/elements_cache.ts:377]: that would be a good log to add in the place that created those changes, not sure it matters here.
anyway, nice to have I guess...
I suggested that because your change makes the original logic (which is hard enough to read as it is) much harder to follow IMO, so much of the code is now actually dedicated to logging and not "real" logic which makes it hard to follow the logic.
so I tried to suggest a way to maybe reduce the amount of code that is required for logging...

**Review by ori-moisis — APPROVED** [review]
